### ISSUE TITLE
IAM permissions error when AwsSecretSource calls secretsmanager:GetSecretValue

### ISSUE THREAD

**Issue by chriskilding** [jenkinsci/aws-secrets-manager-credentials-provider-plugin#74]
(This is a port of 'fake PR https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/pull/65' from @kuntalkumarbasu, who reported this issue before GitHub Issues were enabled for this repository. I'm moving this to an issue so that I don't merge the PR by accident.)

-------

We already have an existing Vault setup to read secrets, we are planning to migrate it to AWS Secrets manager and use this plugin for that purpose. First dual-running the Vault and Secrets Manager providers, and then phasing out Vault.

We have installed the plugin and set up IAM access properly.

We use CASC for the configuration.

There are **no** secrets in Secrets Manager yet.

The vault key setup:

```yaml
credentials:
  system:
    domainCredentials:
      - credentials:
          - string:
              description: 'description
              id: 'ID_TEST'
              scope: GLOBAL
              secret: "${secret_key}"
```

The problem is that when we are running this plugin from CASC via fresh install, it is trying to access an ARN similar to

```
arn:aws:secretsmanager:us-east-2:123456789:secret:ID_TEST-TAhdQc
```

ID_TEST is the key based on which a fictitious ARN is getting searched for, but it does not exist in Secrets Manager yet.

The plugin is throwing a generic error that the jenkins user is `not authorized to perform: secretsmanager:GetSecretValue` on that fictitious ARN, and jenkins is refusing to start.

Basically looks like the plugin is trying to find a ARN for every secret key, may be I am missing something is there any way to avoid this behaviour.

**Comment by kuntalkumarbasu** [comment]
@chriskilding we can close the issue, there was couple of issue, one as you suggested splitting the IAM permission statements did help, apart from that we had some issues with our IAM role setup for jenkins that we fixed as well. Now the plugin is working as expected.

**Comment by chriskilding** [comment]
Great to hear it worked out :)
